libcaf_io/caf/io/middleman.hpp

```diff
@@ -43,7 +43,7 @@ namespace io {
 /// Manages brokers and network backends.
 class middleman : public actor_system::module {
 public:
-  friend class actor_system;
+  friend class ::caf::actor_system;
```

Comment:
Is this change actually needed? `actor_system` is a known type at this point and there should not be any ambiguity.

Reply:
It actually generates a lot of the following tedious warnings.
It may be a bug of the specific clang compile version used by MS (it is a clang 3.8.0)

> 1> ...\actor-framework\libcaf_io\caf/io/middleman.hpp(46,27): warning : unqualified friend declaration referring to type outside of the nearest enclosing namespace is a Microsoft extension; add a nested name specifier [-Wmicrosoft-unqualified-friend]
> 1>    friend class actor_system;
> 1>                       ^
> 1>                       ::caf::
> 1>  1 warning generated.

If you think that such change is dirtier than the problem it wants to solve, I can  squash the other commits ad skip this.

Reply:
If it generates warnings on your platform than it's ok. I just wondered what's the deal here.